contracts/UsulVetoGuard.sol

```diff
@@ -42,49 +42,49 @@ contract UsulVetoGuard is
 
         transferOwnership(_owner);
         vetoVoting = IVetoVoting(_vetoVoting);
-        baseStrategy = IBaseStrategy(_baseStrategy);
+        strategy = IBaseStrategy(_strategy);
         fractalUsul = IFractalUsul(_fractalUsul);
         executionPeriod = _exeuctionPeriod;
 
         emit UsulVetoGuardSetup(
             msg.sender,
             _owner,
             _vetoVoting,
-            _baseStrategy,
+            _strategy,
             _fractalUsul
         );
     }
 
-    /// @notice Queues a transaction for execution
-    /// @param proposalId The ID of the proposal to queue
-    function queueProposal(uint256 proposalId) external {
+    /// @notice Timelocks a transaction for execution
+    /// @param proposalId The ID of the proposal to timelock
+    function timelockProposal(uint256 proposalId) external {
         // If proposal is not yet timelocked, then finalize the strategy
-        if (fractalUsul.state(proposalId) == 0)
-            baseStrategy.finalizeStrategy(proposalId);
+        if (fractalUsul.state(proposalId) == IFractalUsul.ProposalState.ACTIVE)
+            strategy.timelockProposal(proposalId);
 
         require(
-            fractalUsul.state(proposalId) == 2,
-            "Proposal must be timelocked before queuing"
+            fractalUsul.state(proposalId) == IFractalUsul.ProposalState.TIMELOCKED,
+            "Proposal timelock failed"
         );
 
-        (, uint256 timelockDeadline, , ) = fractalUsul.proposals(proposalId);
+        (, uint256 timelockDeadline, , , ) = fractalUsul.getProposal(proposalId);
 
         uint256 executionDeadline = timelockDeadline + executionPeriod;
 
         require(
             block.timestamp > proposals[proposalId].executionDeadline,
-            "Proposal has already been queued"
+            "Proposal has already been timelocked"
         );
 
         proposals[proposalId].executionDeadline = executionDeadline;
-        proposals[proposalId].queuedBlock = block.number;
+        proposals[proposalId].timelockedBlock = block.number;
 
         bytes32[] memory txHashes = fractalUsul.getProposalTxHashes(proposalId);
         for(uint256 i; i < txHashes.length; i++) {
           transactionToProposal[txHashes[i]] = proposalId;
         }
 
-        emit ProposalQueued(msg.sender, proposalId);
+        emit ProposalTimelocked(msg.sender, proposalId);
```

Comment:
minor gas optimization -- cache `txHashes.length` in its own variable before the loop on line 83.
```
uint256 txHashesLength = txHashes.length;
for(uint256 i; i < txHashesLength; i++)
```

Reply:
Hmm just A/B tested this, caching the array length used 5 more gas than the original code